src/QMCHamiltonians/QMCHamiltonian.cpp

```diff
@@ -515,11 +515,9 @@ void QMCHamiltonian::flex_evaluate(const RefVector<QMCHamiltonian>& H_list,
         op.setObservables(ham.Observables);
         op.setParticlePropertyList(pset.PropertyList, ham.myIndex);
       };
-      HC_list[i].get().mw_evaluate(HC_list, P_list);
+      HC_list[0].get().mw_evaluate(HC_list, P_list);
```

Comment:
This is all pretty confusing. That was my thinking documenting via the int what the loop is over, although I grabbed the wrong size. The extract_HC_list is doing what I was thinking of here with the HC_list[i] although with the right limit this should be valid as well. Accessing the mv_evaluate through a particular OperatorBase is just a hack to get at the VTable. 

Anyway this is fine no need to change.

Reply:
HC_list is one component, many walkers not one walker many components. index i is the loop over components. I got 2 walker 5 components and the old code crashes.
Regarding how to access mw_evaluate. Probabaly in Oct. Let us discuss it all the possible solution and find the best one.